app/scripts/controllers/transactions/index.js

```diff
@@ -172,6 +173,7 @@ export default class TransactionController extends EventEmitter {
     this.securityProviderRequest = opts.securityProviderRequest;
     this.messagingSystem = opts.messenger;
     this._hasCompletedOnboarding = opts.hasCompletedOnboarding;
+    this._shouldDisablePublish = opts.shouldDisablePublish;
```

Comment:
For all these new callbacks, is it best we ensure they're optional to minimise the required number of arguments?

We could also group them into an optional `hooks` or 'lifecycle' object for readability?